pkg/httphelper/httphelper.go

```diff
@@ -64,6 +69,18 @@ func IsValidationError(err error) bool {
 	return isJSONErrorWithCode(err, ValidationErrorCode)
 }
 
+// IsRetryableError indicates whether a HTTP request can be safely retried.
+//
+// We consider all postgres errors as retry-able as they usually occur when
+// postgres is read-only (for example during a system update). Data related
+// postgres errors should in general not be retried (because a retry will
+// likely result in the same error), but it is expected that such errors are
+// caught and a validation error returned to the client rather than the
+// postgres error.
+func IsRetryableError(err error) bool {
+	return isJSONErrorWithCode(err, PostgresErrorCode)
```

Comment:
The error response should be marked as retry-able in the JSON message, and we should not send a detailed postgres error (a 500 with the retry flag is sufficient).